src/instructlab/data/generate.py

```diff
@@ -214,10 +222,27 @@ def generate(
         from instructlab.model.backends.llama_cpp import Server as llama_cpp_server
 
         # TODO (cdoern): we really should not edit the cfg object
-        ctx.obj.config.serve.llama_cpp.llm_family = model_family
-        backend_instance = backends.select_backend(
-            cfg=ctx.obj.config.generate.teacher, model_path=model_path
+        gen_cfg = copy.copy(ctx.obj.config)
+        gen_cfg.generate.teacher.llama_cpp.llm_family = model_family
+        if gpus is not None:
+            tps_prefix = "--tensor-parallel-size"
+            if any(
+                s == tps_prefix or s.startswith(tps_prefix + "=")
```

Comment:
This logic now exists here, in vllm, and in evaluate.  vllm has a method (contains_argument) for it or @tiran had suggesting moving to utils